include/cachemere/cache.hpp

```diff
@@ -341,6 +342,41 @@ bool Cache<K, V, I, E, SV, SK, TS>::compare_evict(const K& candidate_key, size_t
     }
 }
 
+template<class K, class V, template<class, class> class I, template<class, class> class E, class SV, class SK, bool TS>
+void Cache<K, V, I, E, SV, SK, TS>::insert_or_update(K&& key, V&& value, size_t key_size, size_t value_size)
+{
+    auto key_and_item = m_data.find(key);
+    if (key_and_item != m_data.end()) {
+        // Update.
+
+        if (value_size > key_and_item->second.m_value_size) {
+            // This item got bigger with the update.
+            m_current_size += value_size - key_and_item->second.m_value_size;
+        } else {
+            // This item got smaller with the update.
+            m_current_size -= key_and_item->second.m_value_size - value_size;
+        }
```

Comment:
why not simply do :

m_current_size -=  value_size;
m_current_size += key_and_item->second.m_value_size;

or something simple like :

m_current_size = m_current_size +  key_and_item->second.m_value_size - value_size;

we update in both branches, so no real optimization in the end of splitting those ...

Reply:
You're right! *I think* this structure is a leftover from times where we took into account the overhead of the policies. Either way, it can definitely be simplified.